ttnn/cpp/ttnn/operations/moreh/moreh_matmul/device/moreh_matmul_program_factory.cpp

```diff
@@ -0,0 +1,520 @@
+
+    const auto reader_kernel_id = tt::operations::primary::CreateReadKernel(
+        program, reader_kernel_file, all_cores, reader_compile_time_args, reader_defines);
+    const auto writer_kernel_id =
+        tt::operations::primary::CreateWriteKernel(program, writer_kernel_file, all_cores, writer_compile_time_args);
+    log_debug(
+        tt::LogOp,
+        "{}:{} DMVK is_dram(input): {}, is_dram(other): {}, is_dram(output): {}",
+        __func__,
+        __LINE__,
+        tt::operations::primary::is_dram(input),
+        tt::operations::primary::is_dram(other),
+        tt::operations::primary::is_dram(output));
+
+    ////////////////////////////////////////////////////////////////////////////
+    //                      ComputeKernel SetUp
+    ////////////////////////////////////////////////////////////////////////////
+    std::map<string, string> compute_defines;
+
+    const auto compute_kernel_file = "ttnn/cpp/ttnn/operations/moreh/moreh_matmul/device/kernels/moreh_matmul.cpp";
+    std::vector<uint32_t> compute_args_group_1 = {
+        num_output_tiles_per_core_group_1,  // num_output_tiles
+        Mt,
+        Nt,
+        Kt,
+        static_cast<uint32_t>(transpose_input),
+        static_cast<uint32_t>(transpose_other),
+        input_mask_h,
+        input_mask_w,
+        other_mask_h,
+        other_mask_w};
+
+    if (bias.has_value()) {
+        compute_defines["FUSE_BIAS"] = "1";
+        compute_args_group_1.push_back(static_cast<uint32_t>(is_scalar_bias));
+    }
+
+    if (fp32_dest_acc_en) {
+        compute_defines["FP32_DEST_ACC_EN"] = "1";
+    }
+
+    const auto compute_kernel_1_id = tt::operations::primary::CreateComputeKernel(
+        program,
+        compute_kernel_file,
+        {core_group_1, num_output_tiles_per_core_group_1, compute_args_group_1},
+        compute_defines,
+        math_fidelity,
+        fp32_dest_acc_en,
+        math_approx_mode);
+
+    std::optional<KernelHandle> compute_kernel_2_id = std::nullopt;
+    if (!core_group_2.ranges().empty()) {
+        std::vector<uint32_t> compute_args_group_2 = {
+            num_output_tiles_per_core_group_2,  // num_output_tiles
+            Mt,
+            Nt,
+            Kt,
+            static_cast<uint32_t>(transpose_input),
+            static_cast<uint32_t>(transpose_other),
+            input_mask_h,
+            input_mask_w,
+            other_mask_h,
+            other_mask_w};
+
+        if (bias.has_value()) {
+            compute_args_group_2.push_back(static_cast<uint32_t>(is_scalar_bias));
+        }
+
+        compute_kernel_2_id = tt::operations::primary::CreateComputeKernel(
+            program,
+            compute_kernel_file,
+            {core_group_2, num_output_tiles_per_core_group_2, compute_args_group_2},
+            compute_defines,
+            math_fidelity,
+            fp32_dest_acc_en,
+            math_approx_mode);
```

Comment:
```suggestion
            math_approx_mode,
            preserve_fp32_precision);
```
Would you consider adding this code, too?

Reply:
I've added the extra options, please check if it is valid